### PR TITLE
support visualizing custom trace events

### PR DESCRIPTION
**Pull request by Vadman97**
## Summary

Supports visualizing trace `metric.name` events as custom trace attributes
for visualizing metrics coming out of a session, such as web vitals of viewport size.

This allows searching for sessions where a given metric value is set.

Closes HIG-4575

## How did you test this change?

Screenshot from 2024-06-24 14-19-40

Screenshot from 2024-06-24 14-21-17

## Are there any deployment considerations?

no

## Does this work require review from our design team?

no

### CODE REVIEW

File: backend/clickhouse/traces.go [hunk 1]
```diff
@@ -79,6 +84,7 @@ var defaultTraceKeys = []*modelInputs.QueryKey{
 	{Name: string(modelInputs.ReservedTraceKeyDuration), Type: modelInputs.KeyTypeNumeric},
 	{Name: string(modelInputs.ReservedTraceKeyParentSpanID), Type: modelInputs.KeyTypeString},
 	{Name: string(modelInputs.ReservedTraceKeySecureSessionID), Type: modelInputs.KeyTypeString},
+	{Name: string(modelInputs.ReservedTraceKeyMetricName), Type: modelInputs.KeyTypeString},
```
Comment (SpennyNDaJets) [inline backend/clickhouse/traces.go:87]: I wasn't able to add metric value on the frontend so curious how you were able to add, or if we should add here?

**Review by SpennyNDaJets — APPROVED** [review]
Click test looks good to me! Just the one question about displaying values

File: backend/clickhouse/traces.go [hunk 2]
```diff
@@ -30,6 +31,7 @@ const TracesSamplingTable = "traces_sampling"
 const TraceKeysTable = "trace_keys"
 const TraceKeyValuesTable = "trace_key_values"
 const TracesByIdTable = "traces_by_id"
+const SelectTraceColumns = "Timestamp, UUID, TraceId, SpanId, ParentSpanId, ProjectId, SecureSessionId, TraceState, SpanName, SpanKind, Duration, ServiceName, ServiceVersion, Environment, HasErrors, TraceAttributes, StatusCode, StatusMessage, Events.Timestamp, Events.Name, Events.Attributes"
```
[the remark this answers is not in the document]
Comment (Vadman97) [reply backend/clickhouse/traces.go:34]: yeah, good point!

File: backend/clickhouse/traces.go (hunk 1 above)
Comment (Vadman97) [reply backend/clickhouse/traces.go:87]: ReservedTraceKeyMetricValue is here already, should match the METRIC_VALUE_COLUMN on the frontend.

File: frontend/src/pages/Traces/TracesList.tsx [hunk 3]
```diff
@@ -173,8 +173,18 @@ export const TracesList: React.FC<Props> = ({
 				},
 			})
 
+			const accessorFn =
```
[the remark this answers is not in the document]
Comment (Vadman97) [reply frontend/src/pages/Traces/TracesList.tsx:176]: yeah, good idea. going to add a ColumnAccessors object here

File: backend/clickhouse/traces.go (hunk 1 above)
Comment (Vadman97) [reply backend/clickhouse/traces.go:88]: Screenshot from 2024-06-25 15-20-04

**Pull request merged**